src/main/java/frc/robot/subsystems/ball/shooter/ShooterImpl.java

```diff
@@ -46,4 +58,15 @@ public void invertPosition(){
         }
         this.shooterSolenoid.set(false);
     }
+
+    public boolean readyForBalls(){
+        if (this.getShooterRPM() > 4000 /* Placholder value */){
```

Comment:
And since this 4000 magic number appears in more than one place, it should definitely be at least extracted to a member variable of the class (or constant). You'll already be doing this once you hook up a SmartTuner though.

Reply:
SmartDashboardTuner*